framework/src/utils/MortarUtils.C

```diff
@@ -262,13 +262,15 @@ projectQPoints3d(const Elem * const msm_elem,
       auto & qp_back = q_pts.back();
       if (primal_elem->type() == TRI3 || primal_elem->type() == TRI6)
       {
-        if (qp_back(0) < 0 || qp_back(1) < 0 || qp_back(0) + qp_back(1) > 1)
+        if (qp_back(0) < -TOLERANCE || qp_back(1) < -TOLERANCE ||
+            qp_back(0) + qp_back(1) > (1 + TOLERANCE))
           mooseException("Quadrature point: ", qp_back, " out of bounds, truncating.");
```

Comment:
Can the quadrature points be forced back on the feasible range if we fall within `TOLERANCE` range?

Reply:
Hmm it's immediately obvious what to do for the 1D comparisons but what about for the 2D comparison ... perhaps just setting it to `qp_back.unit()`

Reply:
I guess my opinion is that if none our later code barfs on these quadrature points within tolerance of the reference domain (but strictly speaking outside of it), that I think it's fine as is

Reply:
@bozeng1997's thermomechanical simulation is pretty neat. With the commit here allowing it to run, here's at the beginning of the sim

<img width="1324" alt="Screen Shot 2023-04-06 at 3 21 03 PM" src="https://user-images.githubusercontent.com/10248304/230504419-57cc23b0-87db-4f06-8ecc-dbb60173104e.PNG">
 
and here's after 100 time steps

<img width="1321" alt="Screen Shot 2023-04-06 at 3 21 14 PM" src="https://user-images.githubusercontent.com/10248304/230504450-83056c9c-d1cc-4a5c-af2a-0d339a30dc30.PNG">

Reply:
Regardless of whether moving the qps back into the strict reference domain or not, I should add a test. @bozeng97 could I use your input file for the test I add to the MOOSE regression suite?

Reply:
I am fine with the code as is too. I do feel like there should be a better way to tackle the variability in problem units, but that's a story for a different day.

Reply:
@lindsayad sure. please use the input file I provided. I am glad that I can help.

Reply:
great, thanks!